terraform/fullnode/gcp/cluster.tf

```diff
@@ -80,4 +103,12 @@ resource "google_container_node_pool" "fullnodes" {
       mode = "GKE_METADATA"
     }
   }
+
+  dynamic "autoscaling" {
+    for_each = var.gke_enable_autoscaling ? [1] : []
```

Comment:
just fyi, the autoscaling here (on the nodepool) has nothing to do with the autoscaling on the cluster level.

The autoscaling on the nodepool enables autoscaling for this specific nodepool whereas the cluster-level one enables nodepool auto provisioning which automatically provisions new nodepools based on workload requirements which then also have autoscaling enabled.

The nodepool auto provisioning is for most practical purposes superior to enabling single nodepool autoscaling.

Imo the best practice here would be to not create a nodepool explicitly at all if nodepool / cluster auto provisioning is enabled (we're doing this internally for the other non node clusters already btw).

Anyways. not 100% required to change this in this PR, but I just wanted to explain it right away and that we should probably just remove the nodepool here altogether.